neo/Consensus/ConsensusContext.cs

```diff
@@ -30,12 +30,18 @@ internal class ConsensusContext : IConsensusContext
         public UInt256[] TransactionHashes { get; set; }
         public Dictionary<UInt256, Transaction> Transactions { get; set; }
         public UInt256[] Preparations { get; set; }
+        public byte[][] PreparationWitnessInvocationScripts { get; set; }
+        public uint[] PreparationTimestamps { get; set; }
```

Comment:
We can store the `PrepareRequest` and `PrepareResponse` payloads directly in the `ConsensusContext`:

```c#
public ConsensusPayload[] PreparationPayloads { get; set; }
```

This combines the three fields `Preparations`, `PreparationWitnessInvocationScripts`, and `PreparationTimestamps`, and the logic is clearer.

Reply:
I'll go ahead and implement that to clean it up.

Reply:
After assessing what is required for this, I think this change should go in a separate PR after merging this one.

Reply:
Is there any difficulty?

Reply:
Yes.

Reply:
We can cache the deserialized `ConsensusMessage` object in `ConsenusPayload`.

```c#
private ConsensusMessage _deserializedMessage = null;
internal T GetDeserializedMessage<T>() where T : ConsensusMessage
{
    if (_deserializedMessage is null)
        _deserializedMessage = ConsensusMessage.DeserializeFrom(Data);
    return (T)_deserializedMessage;
}
```

Reply:
That route will work. I'll re-apply my diff and keep going on it.

Reply:
Thank you for insisting on high standards here, I made the changes and updated the tests, and it looks cleaner.

Reply:
It is finished.